waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -246,3 +246,13 @@ proc dialPeer*(pm: PeerManager, peerInfo: PeerInfo, proto: string, dialTimeout =
     return none(Connection)
 
   return await pm.dialPeer(peerInfo.peerId, peerInfo.addrs, proto, dialTimeout)
+
+proc disconnect*(pm: PeerManager, peerInfo: PeerInfo) : Future[bool] {.async.} =
+  try:
+    let peerId =  peerInfo.peerId
+    debug "Removing peer from manager", peerId = peerInfo.peerId, addr = peerInfo.addrs[0]
```

Comment:
To completely remove the peer from the manager, it would have to be removed from both the in-memory `peerStore` and the persistent `peerStorage` SQLite DB.
That would mean, however, that you would not "remember" a peer's blacklisted status after a restart and would happily reconnect if prompted to. Perhaps it's better to persist the peer's blacklisted-status as an extension to its [`Connectedness` state in the store](https://github.com/status-im/nim-waku/blob/master/waku/v2/node/peer_manager/waku_peer_store.nim#L15)
As it stands though, the peer will simply be reloaded into the store after a restart.

Reply:
Hmm. I get what you mean here.  Currently, the blacklisting is not persistent just stores in an array. On a case of restart, the blacklist data will be lost.

Reply:
I think this could be OK in the interest of getting this PR merged, but let's make sure we capture this in an enhancement issue then.

Reply:
Okay, I'll create a new Issue for this.